tests/integration/butterfree/pipelines/test_feature_set_pipeline.py

```diff
@@ -129,3 +130,179 @@ def test_feature_set_pipeline(
+        create_db_and_table(
+            spark=spark_session,
+            table_reader_id=table_reader_id,
+            table_reader_db=table_reader_db,
+            table_reader_table=table_reader_table,
+        )
+        dbconfig = Mock()
+        dbconfig.get_options = Mock(
+            return_value={
+                "mode": "overwrite",
+                "format_": "parquet",
+                "path": "test_folder/historical/entity/feature_set",
+            }
+        )
+
+        # act
+        test_pipeline = FeatureSetPipeline(
+            source=Source(
+                readers=[
+                    TableReader(
+                        id=table_reader_id,
+                        database=table_reader_db,
+                        table=table_reader_table,
+                    ).with_incremental_strategy(
+                        incremental_strategy=IncrementalStrategy(column="timestamp")
+                    ),
+                ],
+                query=f"select * from {table_reader_id} ",  # noqa
+            ),
+            feature_set=FeatureSet(
+                name="feature_set",
+                entity="entity",
+                description="description",
+                features=[
+                    Feature(
+                        name="feature",
+                        description="test",
+                        transformation=SparkFunctionTransform(
+                            functions=[
+                                Function(F.avg, DataType.FLOAT),
+                                Function(F.stddev_pop, DataType.FLOAT),
+                            ],
+                        ).with_window(
+                            partition_by="id",
+                            order_by=TIMESTAMP_COLUMN,
+                            mode="fixed_windows",
+                            window_definition=["1 day"],
+                        ),
+                    ),
+                ],
+                keys=[
+                    KeyFeature(
+                        name="id",
+                        description="The user's Main ID or device ID",
+                        dtype=DataType.INTEGER,
+                    )
+                ],
+                timestamp=TimestampFeature(),
+            ),
+            sink=Sink(writers=[HistoricalFeatureStoreWriter(db_config=dbconfig)],),
+        )
+        test_pipeline.run(start_date="2016-04-12", end_date="2016-04-13")
+
+        # assert
+        path = dbconfig.get_options("historical/entity/feature_set").get("path")
+        df = spark_session.read.parquet(path).orderBy(TIMESTAMP_COLUMN)
+
+        target_df = fixed_windows_output_feature_set_date_dataframe.orderBy(
+            test_pipeline.feature_set.timestamp_column
+        )
+
+        # assert
+        assert_dataframe_equality(df, target_df)
+
+        # tear down
+        shutil.rmtree("test_folder")
+
+    def test_feature_set_pipeline_with_execution_date(
```

Comment:
Same points here in this test:
- About the use of file system
- Here you assert the same target dataframe from the other test, but here the user is really just passing one date (12/04) for the execution.

Reply:
Regarding the target dataframe, it was just for convenience, since both causes would produce the same results.